crates/oci/src/client.rs

```diff
@@ -669,6 +712,32 @@ impl Client {
     }
 }
 
+struct ComponentSourceLoader;
+
+#[async_trait]
+impl spin_compose::ComponentSourceLoader for ComponentSourceLoader {
```

Comment:
This seems to duplicate the `trigger` crate ComponentSourceLoader.  Can we unify them?

Reply:
Unified the loaders into a `spin_compose::ComponentSourceLoaderFs`